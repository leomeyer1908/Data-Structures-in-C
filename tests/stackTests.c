#include <stdio.h>
#include <stdint.h>
#include "../Stack.h"

int main() {
    Stack myStack;
    initStack(&myStack);

    printf("Test Case: Empty Stack\n");
    printf("Top of stack: %zu\n", (size_t) topStack(&myStack));

    printf("\nTest Case: Pushing elements to stack\n");
    printf("Pushing 10\n");
    pushStack(&myStack, (void*) (uintptr_t) 10);
    printf("Top of stack: %zu\n", (size_t) topStack(&myStack));
    printf("Pushing 20\n");
    pushStack(&myStack, (void*) (uintptr_t) 20);
    printf("Top of stack: %zu\n", (size_t) topStack(&myStack));
    printf("Pushing 30\n");
    pushStack(&myStack, (void*) (uintptr_t) 30);
    printf("Top of stack: %zu\n", (size_t) topStack(&myStack));

    printf("\nTest Case: Popping elements from the stack\n");
    printf("Popping:\n");
    popStack(&myStack);
    printf("Top of stack: %zu\n", (size_t) topStack(&myStack));
    printf("Popping:\n");
    popStack(&myStack);
    printf("Top of stack: %zu\n", (size_t) topStack(&myStack));
    printf("Popping:\n");
    popStack(&myStack);
    printf("Top of stack: %zu\n", (size_t) topStack(&myStack));

    printf("\nTest Case: Popping elements from an empty stack\n");
    printf("Popping:\n");
    popStack(&myStack);
    printf("Top of stack: %zu\n", (size_t) topStack(&myStack));


    printf("\nTest Case: Pushing and popping elements to stack again\n");
    printf("Pushing 10\n");
    pushStack(&myStack, (void*) (uintptr_t) 10);
    printf("Top of stack: %zu\n", (size_t) topStack(&myStack));
    printf("Pushing 20\n");
    pushStack(&myStack, (void*) (uintptr_t) 20);
    printf("Top of stack: %zu\n", (size_t) topStack(&myStack));
    printf("Pushing 30\n");
    pushStack(&myStack, (void*) (uintptr_t) 30);
    printf("Top of stack: %zu\n", (size_t) topStack(&myStack));
    printf("Popping:\n");
    popStack(&myStack);
    printf("Top of stack: %zu\n", (size_t) topStack(&myStack));
    printf("Popping:\n");
    popStack(&myStack);
    printf("Top of stack: %zu\n", (size_t) topStack(&myStack));
    printf("Popping:\n");
    popStack(&myStack);
    printf("Top of stack: %zu\n", (size_t) topStack(&myStack));
    
    printf("\nTest Case: Destroy Stack:\n");
    destroyStack(&myStack);
    printf("Top of stack: %zu\n", (size_t) topStack(&myStack));

    printf("\nTest Case: Pushing and popping elements to stack last time\n");
    printf("Pushing 10\n");
    pushStack(&myStack, (void*) (uintptr_t) 10);
    printf("Top of stack: %zu\n", (size_t) topStack(&myStack));
    printf("Pushing 20\n");
    pushStack(&myStack, (void*) (uintptr_t) 20);
    printf("Top of stack: %zu\n", (size_t) topStack(&myStack));
    printf("Pushing 30\n");
    pushStack(&myStack, (void*) (uintptr_t) 30);
    printf("Top of stack: %zu\n", (size_t) topStack(&myStack));
    printf("Popping:\n");
    popStack(&myStack);
    printf("Top of stack: %zu\n", (size_t) topStack(&myStack));
    printf("Popping:\n");
    popStack(&myStack);
    printf("Top of stack: %zu\n", (size_t) topStack(&myStack));
    printf("Popping:\n");
    popStack(&myStack);
    printf("Top of stack: %zu\n", (size_t) topStack(&myStack));

    return 0;
}