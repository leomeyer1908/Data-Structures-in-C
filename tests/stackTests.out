Test Case: Empty Stack
Top of stack: 0

Test Case: Pushing elements to stack
Pushing 10
Top of stack: 10
Pushing 20
Top of stack: 20
Pushing 30
Top of stack: 30

Test Case: Popping elements from the stack
Popping:
Top of stack: 20
Popping:
Top of stack: 10
Popping:
Top of stack: 0

Test Case: Popping elements from an empty stack
Popping:
Top of stack: 0

Test Case: Pushing and popping elements to stack again
Pushing 10
Top of stack: 10
Pushing 20
Top of stack: 20
Pushing 30
Top of stack: 30
Popping:
Top of stack: 20
Popping:
Top of stack: 10
Popping:
Top of stack: 0

Test Case: Destroy Stack:
Top of stack: 0

Test Case: Pushing and popping elements to stack last time
Pushing 10
Top of stack: 10
Pushing 20
Top of stack: 20
Pushing 30
Top of stack: 30
Popping:
Top of stack: 20
Popping:
Top of stack: 10
Popping:
Top of stack: 0
